src/providers/token-fee-fetcher.ts

```diff
@@ -0,0 +1,103 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { ChainId } from '@uniswap/sdk-core';
+
+import { TokenFeeDetector__factory } from '../types/other/factories/TokenFeeDetector__factory';
+import { log, WRAPPED_NATIVE_CURRENCY } from '../util';
+
+import { IMulticallProvider } from './multicall-provider';
+import { ProviderConfig } from './provider';
+
+type Address = string;
+
+export type TokenFeeResult = {
+  buyFeeBps: BigNumber;
+  sellFeeBps: BigNumber;
+};
+export type TokenFeeMap = Record<Address, TokenFeeResult>;
+
+// address at which the FeeDetector lens is deployed
+const FEE_DETECTOR_ADDRESS = '0x57eC54d113719dDE9A90E6bE807524a86560E89D';
```

Comment:
We'll need this per chain eventually, it seems like we will start adding V2 support soon-ish per governance vote.

Reply:
got it, we already pass chain id into constructor. We can make FEE_DETECTOR_ADDRESS a map of `ChainId to deployer address